File: app/src/main/java/com/art2cat/dev/moonlightnote/controller/moonlight/TrashFragment.java
package com.art2cat.dev.moonlightnote.controller.moonlight;

import com.google.firebase.database.DatabaseReference;
import com.google.firebase.database.Query;

/**
 * Created by Art on 2016/11/2 20:06.
 */

public class TrashFragment extends MoonlightListFragment {

  @Override
  public Query getQuery(DatabaseReference databaseReference) {
    return databaseReference.child("users-moonlight")
        .child(getUid()).child("trash");
  }

  @Override
  public boolean isTrash() {
    return true;
  }
}

File: app/src/main/java/com/art2cat/dev/moonlightnote/controller/moonlight/TrashFragment.kt
package com.art2cat.dev.moonlightnote.controller.moonlight

import com.google.firebase.database.DatabaseReference
import com.google.firebase.database.Query

/**
 * Created by Rorschach
 * on 24/05/2017 8:08 PM.
 */

class TrashFragment : MoonlightListFragment() {
    override fun getQuery(databaseReference: DatabaseReference): Query? {
        return databaseReference.child("users-moonlight")
                .child(uid).child("trash")
    }

    override val isTrash: Boolean
        get() = true
}
